flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/planner/plan/rules/physical/stream/StreamPhysicalPythonGroupWindowAggregateRule.java

```diff
@@ -78,11 +78,11 @@ public boolean matches(RelOptRuleCall call) {
                         .anyMatch(x -> PythonUtil.isPythonAggregate(x, PythonFunctionKind.PANDAS));
         boolean existJavaFunction =
                 aggCalls.stream().anyMatch(x -> !PythonUtil.isPythonAggregate(x, null));
+        if (existPandasFunction && existGeneralPythonFunction) {
+            throw new TableException(
+                    "Pandas UDAFs and General Python UDAFs are not supported in used together currently.");
```

Comment:
Question: so Pandas UDAF and General UDAF cannot work together because one input is Row and another is non-row?

Reply:
The serializer used by general UDF and pandas cannot be easily used in the same data channel. So for `General UDF` and `Pandas UDF`, we added a special logical rule(`PythonCalcSplitPandasInProjectionRule`) to separate them into the different operator, but for `UDAF`, we can't separate these `UDAF`s into different operators.